libc-bottom-half/cloudlibc/src/libc/sys/wasi/preview2.c

```diff
@@ -0,0 +1,4448 @@
+// Generated by `wit-bindgen` 0.16.0. DO NOT EDIT!
```

Comment:
I'd like to avoid adding new code within the `cloudlibc` directory, so that `cloudlibc` just contains code derived from cloudlibc. Could you move the bindings to a directory like libc-bottom-half/sources/preview2, or even just libc-bottom-half/preview2?

Reply:
Sure, no problem.  We also added various new files under the `cloudlibc` directory in https://github.com/WebAssembly/wasi-libc/pull/449/files, while also adding code to other existing files.  Should we move all the new files elsewhere, and is there anything special we should do when modifying the existing files?

Reply:
Yeah, ideally we should move those new files out of `cloudlibc`. I don't think we need to do anything else special though.

We already only use a few small parts of the full `cloudlibc`, and the parts we do use are pretty specific to preview1 which was derived from cloudabi, so as preview2 progresses, we may even stop using `cloudlibc` altogether.

Reply:
I've moved the .c and .o files to libc-bottom-half/sources